File: app/src/main/java/com/brainwellnessspa/DashboardOldModule/Appointment/AppointmentDetails/AptBookletFragment.java
package com.brainwellnessspa.DashboardOldModule.Appointment.AppointmentDetails;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.net.Uri;
import android.os.Bundle;

import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.brainwellnessspa.BWSApplication;
import com.brainwellnessspa.DashboardOldModule.Models.AppointmentDetailModel;
import com.brainwellnessspa.R;
import com.brainwellnessspa.Utility.CONSTANTS;
import com.brainwellnessspa.databinding.FragmentAptBookletBinding;
import com.segment.analytics.Properties;

public class AptBookletFragment extends Fragment {
    FragmentAptBookletBinding binding;
    AppointmentDetailModel.ResponseData appointmentDetail;
    String UserID;
    Properties p;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_apt_booklet, container, false);
        View view = binding.getRoot();
        SharedPreferences shared1 = getActivity().getSharedPreferences(CONSTANTS.PREF_KEY_LOGIN, Context.MODE_PRIVATE);
        UserID = (shared1.getString(CONSTANTS.PREF_KEY_UserID, ""));
        if (getArguments() != null) {
            appointmentDetail = getArguments().getParcelable("AppointmentDetail");
        }
        binding.tvTilte.setText(appointmentDetail.getName());
        binding.btnComplete.setOnClickListener(view1 -> {
            Intent i = new Intent(Intent.ACTION_VIEW);
            i.setData(Uri.parse(appointmentDetail.getBooklet()));
            startActivity(i);
            BWSApplication.showToast("Complete the booklet", getActivity());
            p = new Properties();
            p.putValue("userId", UserID);
            p.putValue("sessionId", appointmentDetail.getId());
            p.putValue("sessionName", appointmentDetail.getName());
            p.putValue("bookletUrl", appointmentDetail.getBookUrl());
            BWSApplication.addToSegment("Complete Booklet Clicked", p, CONSTANTS.track);
        });
        return view;
    }
}

File: app/src/main/java/com/brainwellnessspa/dashboardOldModule/appointment/appointmentDetails/AptBookletFragment.kt
package com.brainwellnessspa.dashboardOldModule.appointment.appointmentDetails

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import com.brainwellnessspa.BWSApplication
import com.brainwellnessspa.R
import com.brainwellnessspa.dashboardOldModule.models.AppointmentDetailModel
import com.brainwellnessspa.databinding.FragmentAptBookletBinding
import com.brainwellnessspa.utility.CONSTANTS
import com.segment.analytics.Properties

class AptBookletFragment : Fragment() {
    lateinit var binding: FragmentAptBookletBinding
    private var appointmentDetail: AppointmentDetailModel.ResponseData? = null
    var userId: String? = null
    var p: Properties? = null
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_apt_booklet, container, false)
        val view = binding.root
        val shared1 = requireActivity().getSharedPreferences(CONSTANTS.PREF_KEY_LOGIN, Context.MODE_PRIVATE)
        userId = shared1.getString(CONSTANTS.PREF_KEY_UserID, "")
        if (arguments != null) {
            appointmentDetail = requireArguments().getParcelable("AppointmentDetail")
        }
        binding.tvTilte.text = appointmentDetail!!.name
        binding.btnComplete.setOnClickListener {
            val i = Intent(Intent.ACTION_VIEW)
            i.data = Uri.parse(appointmentDetail!!.booklet)
            startActivity(i)
            BWSApplication.showToast("Complete the booklet", activity)
            /*p = Properties()
            p!!.putValue("sessionId", appointmentDetail!!.id)
            p!!.putValue("sessionName", appointmentDetail!!.name)
            p!!.putValue("bookletUrl", appointmentDetail!!.bookUrl)
            BWSApplication.addToSegment("Complete Booklet Clicked", p, CONSTANTS.track)*/
        }
        return view
    }
}